lib/src/main/java/ECSE428Project/service/LoginService.java

```diff
@@ -0,0 +1,68 @@
+package ECSE428Project.service;
+
+import ECSE428Project.dao.AccountRepository;
+import ECSE428Project.model.Account;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import org.springframework.web.server.ResponseStatusException;
+
+import java.util.List;
+
+import static org.hibernate.internal.util.collections.ArrayHelper.toList;
+
+
+@Service
+public class LoginService {
+
+    @Autowired
+    private AccountRepository accountRepository;
+
+    @Transactional
+    public Account profileLogin(String email, String password) {
+
+        if (email == null || password == null) {
+            throw new ResponseStatusException(HttpStatus.BAD_REQUEST, "Credentials cananot be empty! Please Enter Valid Credentials");
+        }
+
+        Account profile = accountRepository.findAccountByEmail(email);
+
+        if (profile == null) {
+            throw new ResponseStatusException(HttpStatus.NOT_FOUND, "No account assosciated to the provided credentials");
+        }
+
+        // Checking if password is correct and account isn't logged in
+        if (password.equals(profile.getPassword()) && !profile.isLoggedIn()) {
+            profile.setLoggedIn(true);
+            accountRepository.save(profile);
+            return profile;
+
+        } else if (profile.isLoggedIn()) {
+            throw new IllegalArgumentException("Already Logged In");
+        } else if (!password.equals(profile.getPassword())) {
+            throw new IllegalArgumentException("Password did not match records");
+        }
+        else {
+            return null;
+        }
+    }
+
+
+
+    @Transactional
+    public Account getLoggedInUser(){
+        // Get all profiles in database and check which one is logged in
+        List<Account> allProfiles = toList(accountRepository.findAll());
+        for (Account profile : allProfiles) {
+            if (profile.isLoggedIn()) {
+                return profile;
+            }
+        }
+        throw new IllegalArgumentException("Not Logged In");
```

Comment:
Why is an exception thrown when no one is logged in. Also what is the use of this method since it only finds any logged in account and not a specific one based on the id or email?.

Reply:
Fixed in f1a72cd